app/controllers/impersonate_controller.rb

```diff
@@ -79,18 +84,21 @@ def check_if_user_impersonateable
       end
     else
       unless params[:impersonate][:name].empty?
-        # E1991 : check whether instructor is currently in anonymized view
         overwrite_session
       end
     end
   end
 
   # Function to display appropriate error messages based on different username provided, the message explains each error
+  # This function checks params values and displays error_message based on the user name .This is initial check to see if username is valid
+
   def display_error_msg
```

Comment:
We'll be working to eliminate this method.

Reply:
should we do anything to this method?

Reply:
By "we" I meant your team and I xD

Reply:
Removed this method and its dependencies in the code